### PR TITLE
feat: First stable version

### PR DESCRIPTION
**Pull request by talend-glorieux**
* First stable version of the library.
* Fixes integration issues between react-jsonschema-form and
react-bootstrap
* Fixes react-jsonschema-form onChange target
* Adds obejct-rest-spread

### CODE REVIEW

File: src/Checkbox.js [hunk 1]
```diff
@@ -4,13 +4,13 @@ import FormGroup from 'react-bootstrap/lib/FormGroup';
 import ControlLabel from 'react-bootstrap/lib/ControlLabel';
 import BCheckbox from 'react-bootstrap/lib/Checkbox';
 
-const Checkbox = (props) => <FormGroup>
-	<ControlLabel>{props.name}</ControlLabel>
+const Checkbox = ({ name, formData, schema, required, onChange }) => <FormGroup>
```
Comment (jmfrancois) [inline src/Checkbox.js]: I like the destructuring of props !
Just on the form: I would prefer following alphabetic order and synchronize with proptypes

File: src/Form.js [hunk 2]
```diff
@@ -5,34 +5,18 @@ import Button from 'react-bootstrap/lib/Button';
...
-}
+const Form = (props) => <RJSForm
+	schema={props.schema}
+	formData={props.formData}
+	onSubmit={props.onSubmit}
+	fields={{ SchemaField }}
+>
+	<div>
+		<Button bsStyle="primary" type="submit">
+			<i className="fa icon-edit" />Save
```
Comment (jmfrancois) [inline src/Form.js]: fa fa-pencil.
We must align on icon library

File: src/SchemaField.js [hunk 3]
```diff
@@ -1,21 +1,21 @@
...
+	<ControlLabel>{name}</ControlLabel>
 	<FormControl
-		type={props.type}
-		value={props.formData || props.schema.default}
-		required={props.required}
-		onChange={props.onChange}
+		type={type}
+		value={formData || schema.default}
+		required={required}
+		onChange={(e) => onChange(e.target.value)}
```
Comment (jmfrancois) [inline src/SchemaField.js]: onChange props is not required.
I guess this throw an exception if the props is missing

File: src/SchemaField.js [hunk 4]
```diff
@@ -34,23 +34,20 @@ Input.propTypes = {
...
+const Dropdown = ({ name, onChange, schema }) => <FormGroup>
+	<ControlLabel>{name}</ControlLabel>
 	<FormControl
 		componentClass="select"
-		label={props.name}
-		onChange={props.onChange}
-		value={props.formData || props.schema.default}
-	/>
+		label={name}
+		onChange={(e) => onChange(e.target.value)}
```
Comment (jmfrancois) [inline src/SchemaField.js]: same here

File: src/ShowIf.js [hunk 5]
```diff
@@ -16,9 +16,9 @@ class ShowIf extends React.Component {
 	//}
 
 	handleChange() {
-		const p = Object.assign({}, this.props);
-		p.schema = retrieveSchema(this.props.schema, this.props.registry.definitions);
-		this.props.onSchemaChange(p.schema);
+		//const p = Object.assign({}, this.props);
+		//p.schema = retrieveSchema(this.props.schema, this.props.registry.definitions);
+		//this.props.onSchemaChange(p.schema);
 	}
```
Comment (jmfrancois) [inline src/ShowIf.js]: Could you explain me what happens here  ?

File: src/ShowIf.js [hunk 6]
```diff
@@ -16,9 +16,9 @@ class ShowIf extends React.Component {
 	//}
 
 	handleChange() {
-		const p = Object.assign({}, this.props);
-		p.schema = retrieveSchema(this.props.schema, this.props.registry.definitions);
-		this.props.onSchemaChange(p.schema);
+		//const p = Object.assign({}, this.props);
```
Comment (acateland) [inline src/ShowIf.js]: why keeping something comented ?

File: src/SchemaField.js [hunk 7]
```diff
@@ -62,11 +59,19 @@ const fieldsMap = {
...
 	constructor(props) {
+		if (props.schema && Object.keys(props.schema).length === 0) {
+			throw new SchemaException('You must provide a schema');
+		}
 		super(props);
-		this.handleSchemaChange.bind(this);
+		this.handleSchemaChange = this.handleSchemaChange.bind(this);
 	}
 
 	handleSchemaChange(schema) {
```
Comment (acateland) [inline src/SchemaField.js]: why keeping this ?

File: src/SchemaField.js (hunk 3 above)
Comment (talend-glorieux) [inline src/SchemaField.js]: It's actually required by the react-jsonschema-form library. I need to remap the onChange call to make sure it targets the right property.

File: src/Form.js (hunk 2 above)
Comment (talend-glorieux) [inline src/Form.js]: Good catch ;)

**Comment by talend-glorieux** [comment]
I've removed conditional display for the time being so I can make the first release.

File: src/ShowIf.js (hunk 6 above)
Comment (talend-glorieux) [inline src/ShowIf.js]: Removed.

File: src/ShowIf.js (hunk 5 above)
Comment (talend-glorieux) [inline src/ShowIf.js]: I've removed that part for the time being.

**Comment by talend-glorieux** [comment]
@acateland @jmfrancois  Could you double check that PR?

File: src/Checkbox.js [hunk 8]
```diff
@@ -4,22 +4,25 @@ import FormGroup from 'react-bootstrap/lib/FormGroup';
...
-const Checkbox = (props) => <FormGroup>
-	<ControlLabel>{props.name}</ControlLabel>
+const Checkbox = ({ formData, name, onChange, required, schema  }) => <FormGroup>
+	<ControlLabel>{name}</ControlLabel>
 	<BCheckbox
-		checked={props.formData || props.schema.default}
-		label={props.name}
-		required={props.required}
-		onChange={props.onChange}
+		checked={typeof formData === "undefined" ? false : formData }
```
Comment (frassinier) [inline src/Checkbox.js]: Is this mandatory? Something like `checked={ formData || false }` could not work?

File: src/Form.js [hunk 9]
```diff
@@ -9,36 +9,36 @@ class Form extends React.Component {
...
 		return (<RJSForm
 			schema={this.props.schema}
-			formData={this.props.formData || this.state.formData}
+			formData={this.props.formData}
 			onSubmit={this.props.onSubmit}
-			onChange={this.props.onChange || this.handleChange}
 			fields={{ SchemaField }}
+			onChange={this.handleSchemaChange}
 		>
 			<div>
```
Comment (frassinier) [inline src/Form.js]: Is that `<div>` really useful?

File: src/Checkbox.js (hunk 8 above)
Comment (talend-glorieux) [inline src/Checkbox.js]: Easier indeed ;)

File: src/Form.js (hunk 9 above)
Comment (talend-glorieux) [inline src/Form.js]: No. Removing it.

File: src/Form.js [hunk 10]
```diff
@@ -9,36 +9,34 @@ class Form extends React.Component {
...
 		super(props);
-		this.state = { formData: {} };
-		this.handleChange = this.handleChange.bind(this);
+		this.handleSchemaChange = this.handleSchemaChange.bind(this);
 	}
 
-	handleChange(change) {
-		this.setState({ formData: change.formData });
+	handleSchemaChange(change) {
+		console.debug('Schema Change', change);
```
Comment (jmfrancois) [inline src/Form.js]: console.debug !

File: src/Form.js [hunk 11]
```diff
@@ -9,36 +9,33 @@ class Form extends React.Component {
...
 	constructor(props) {
 		super(props);
-		this.state = { formData: {} };
-		this.handleChange = this.handleChange.bind(this);
+		this.handleSchemaChange = this.handleSchemaChange.bind(this);
 	}
 
-	handleChange(change) {
-		this.setState({ formData: change.formData });
+	handleSchemaChange(change) {
```
Comment (acateland) [inline src/Form.js]: empty handler ?

**Pull request merged**